libraries/AP_Mount/AP_Mount_SToRM32_serial.cpp

```diff
@@ -71,31 +69,39 @@ void AP_Mount_SToRM32_serial::update()
         // RC radio manual angle control, but with stabilization from the AHRS
         case MAV_MOUNT_MODE_RC_TARGETING: {
             // update targets using pilot's RC inputs
-            MountTarget rc_target {};
-            if (get_rc_rate_target(rc_target)) {
-                update_angle_target_from_rate(rc_target, _angle_rad);
-            } else if (get_rc_angle_target(rc_target)) {
-                _angle_rad = rc_target;
+            if (get_rc_rate_target(mnt_target.rate_rads)) {
+                mnt_target.target_type = MountTargetType::RATE;
+                update_angle_target_from_rate(mnt_target.rate_rads, mnt_target.angle_rad);
+            } else {
+                mnt_target.target_type = MountTargetType::ANGLE;
+                IGNORE_RETURN(get_rc_angle_target(mnt_target.angle_rad));
             }
+            mnt_target.angle_rad.yaw_is_ef = false;
             resend_now = true;
             break;
         }
 
         // point mount to a GPS point given by the mission planner
         case MAV_MOUNT_MODE_GPS_POINT:
-            if (get_angle_target_to_roi(_angle_rad)) {
+            if (get_angle_target_to_roi(mnt_target.angle_rad)) {
+                mnt_target.target_type = MountTargetType::ANGLE;
+                mnt_target.angle_rad.yaw_is_ef = false;
                 resend_now = true;
             }
             break;
 
         case MAV_MOUNT_MODE_HOME_LOCATION:
-            if (get_angle_target_to_home(_angle_rad)) {
+            if (get_angle_target_to_home(mnt_target.angle_rad)) {
+                mnt_target.target_type = MountTargetType::ANGLE;
+                mnt_target.angle_rad.yaw_is_ef = false;
```

Comment:
as mentioned above, this external setting of the yaw_is_ef doesn't look correct to me.